### PR TITLE
Recalculate cart totals when removing coupon

### LINKED ISSUE #260: removeCoupons mutation not updating cart totals
**Describe the bug**
I'm having an issue with the removeCoupons mutations. In the cart response the totals are not reflecting the changes from removing the coupon. Coupon is being removed from _appliedCoupons_ but all cart totals and taxes are not being recalculated.

**To Reproduce**
1. Add a coupon with applyCoupon
2. Remove a couon with removeCoupons

**Expected behavior**
cart object should contain recalculated totals

### PR DESCRIPTION
**Pull request by zamson**
### Your checklist for this pull request
Thanks for sending a pull request! Please make sure you click the link above to view the contribution guidelines, then fill out the blanks below.

🚨Please review the [guidelines for contributing](./CONTRIBUTING.md) to this repository.

- [ ] Make sure you are making a pull request against the **develop branch** (left side). Also you should start *your branch* off *our develop*.
- [ ] Make sure you are requesting to pull request from a **topic/feature/bugfix branch** (right side). Don't pull request from your master!

What does this implement/fix? Explain your changes.
---------------------------------------------------
Recalculate cart totals when removing a coupon.

Does this close any currently open issues?
------------------------------------------
Closes #260 

Any relevant logs, error output, GraphiQL screenshots, etc?
-------------------------------------
(If it’s long, please paste to https://ghostbin.com/ and insert the link here.)

Any other comments?
-------------------
…

Where has this been tested?
---------------------------
**Operating System:** …

**WordPress Version:** …

### CODE REVIEW

**Comment by kidunot89** [comment]
I think this is a much deeper problem. Recalculation is actually done by WooCommerce during the `WC()->cart->remove_coupon()`.

File: includes/mutation/class-cart-remove-coupons.php [hunk 1]
```diff
@@ -85,6 +85,9 @@ public static function mutate_and_get_payload() {
 				}
 			}
 
+			// Recalculate totals.
+			\WC()->cart->calculate_totals();
```
Comment (kidunot89) [inline includes/mutation/class-cart-remove-coupons.php:89]: I think this is a much deeper problem. Cart total recalculation is actually done by WooCommerce during the `WC()->cart->remove_coupon()`.

Comment (zamson) [reply includes/mutation/class-cart-remove-coupons.php:89]: 🤔Okay. Are you referring to refresh totals? https://woocommerce.wp-a2z.org/oik_api/wc_cartremove_coupon/

Comment (kidunot89) [reply includes/mutation/class-cart-remove-coupons.php:89]: Yes. This issue could also be an issue how the `cart` is being resolved as well.

**Comment by kidunot89** [comment]
@zamson Could you rebase this from the `release/v0.5.1` branch and change the PR's target to `release/v0.5.1` so we can test that this is working.

File: includes/mutation/class-cart-remove-coupons.php (hunk 1 above)
Comment (kidunot89) [reply includes/mutation/class-cart-remove-coupons.php:89]: @zamson can you check and see if this issue still persist on [Release v0.5.1](https://github.com/wp-graphql/wp-graphql-woocommerce/releases/tag/v0.5.1)

Comment (zamson) [reply includes/mutation/class-cart-remove-coupons.php:89]: Yes it's still there.

**Pull request closed without merging**